Issue127

shigdel opened the pull request:
renamed and changed the content   CalculatedUnifrac to CalculatedUnifrac

shigdel committed:
Update calculateUniFrac.R

shigdel committed:
Merge pull request #1 from shigdel/shigdel-patch-1

Update calculateUniFrac.R

shigdel committed:
renamed calculatedUnifrac

ChouaibB reviewed (changes requested):
File should be renamed as "calculateUnifrac"

TuomasBorman reviewed (changes requested):
Hi, thanks for the PR!

Now you have two pull requests open for this same thing. In addition to this, you have opened PRs https://github.com/microbiome/mia/pull/153 and https://github.com/microbiome/mia/pull/129 (already closed) which are related to the issue https://github.com/microbiome/mia/issues/127

It is better to have one PR for one thing to simplify things. Because now some discussions are in  another PR, and this is some kind of documentation about what we have done and why we haven't done some things. 

If you want to revert / undo things that you have made to your branch, that is always possible

**So, for the future PRs: one issue = one PR is better**

Also, you are merging this to `microbiome:issue127` branch and not to `master`. Is there any reason for that? **I think you should merge this directly into the master.** 

That can be easily changed: 

1. press edit (next to the PR title)
2. change base: issue127 to master

**You haven't generated manual page based on your changes. So you can do that by running `devtools::document()`**, to see if there is any errors.

Otherwise looks good

-Tuomas

antagomir commented:
I would support that only specific core maintainers will merge PRs to master.

TuomasBorman commented:
> I would support that only specific core maintainers will merge PRs to master.

Well that might be actually better for example because of the limitations of testing when project is forked

FelixErnst commented:
Please add a new generic for the old name, add a deprected note, specifically explaining the tiny renaming part. Thanks

antagomir commented:
@shigdel let us know if there are any problems with implementing the proposed changes -

TuomasBorman committed:
Issue127 suggestions (#2)

* Update calculateUniFrac.R

changed UniFrac TO Unifrac

* Reverted changes related to renaming of calculateUniFrac

* Deprecated calculateDistance (#126)

* Removed calculateDistance from exports

* Version bump

* Minor change

* Revert changes related to removing calculateDistance

* Deprecated calculateDistance

* Fix

* Fixed a warning message

* meltAssay fix (colData is not added) (#130)

* meltAssay fix, check.names = FALSE

* Fix

* Added check_names argument

* Fix

* Fix

* meltAssay: parameter assay_name --> abund_values

* Fixed add_col_data

* Fix sample name mismatch

* Added tests, version bump

* Fix bug in loadFromQIIME2()

loadFromQIIME2() now checks that the sample ids in the supplied sample
metadata file are compatible with those in the supplied feature table
file, and align them if necessary.

* Fixed meltAssay, add_col_data was not working with one column (#131)

* keep dimnames of feature table consistent with meta data, #132 (#134)

* Import functions fix (#135)

* Fixed bugs that occur when all data is not provided

* Disabled subsetting of rowData in makePseqFromTSE

* Added colnames and rownames to empty DataFrames

* Added name order check

* Reverted changes regarding subsetting of rowData

* Fix

* Removed name order check from loadFromBiom

* Version bump

* ok

* Minor change to example

* draft for subsample

test not yet added

* draft for getSubsample

added test and example in vignette.

* fixes

* name change

* makePseqFromTreeSE, factors in taxonomy table converted to characters (#145)

* makePseqFromTreeSE, factors in taxonomy table converted to characters

* Added test

* Biocfixes (#143)

* sync

* Several check and BiocCheck issues solved

* donttest to dontrun

* Vignettefix (#142)

* sync

* vignette fix

* agglomerateByRank: warning when binary or negative values (#148)

* Update subsampleCounts.R

* return_type option added

Along with updated tests and vignette sections for subsampleCounts.

* fix build

* check for `return_type` arg plus version bump

* changes

basically improve checks

* Melt assay matrix fix (#152)

* meltAssay fix (duplicated rownames) (#157)

* Modify duplicated rownames

* Use getTaxonomyLabels instead of make.names

* Added warning about making rownames unique

* getTaxonomyLabels -> make.unique, version bump, NEWS update

* mergeCols fix (#156)

Co-authored-by: FelixErnst <felix.gm.ernst@outlook.com>

* only tse

* related to #112

related to #112
news, version-up, man-page, description

* clean

* Delete mia.Rproj

* exprs_values -> abund_values (#159)

* exprs_values -> abund_values

* Rephrased abund_values man page changes

* cleanup

* Update subsampleCounts.R

* cosmetic

* bump x.y.z version to even y prior to creation of RELEASE_3_14 branch

* bump x.y.z version to odd y following creation of RELEASE_3_14 branch

* Delete test-5calculateDistance.R

* Added getExperimentCrossCorrelation function (#149)

Co-authored-by: FelixErnst <felix.gm.ernst@outlook.com>

* getExperimentCrossCorrelation bugfix

* Version bump

* test phylo trees: expect_equal --> expect_identical (#164)

Fixes https://github.com/microbiome/mia/issues/162

* Added calculateOverlap (#150)

* Implemented microbiome::overlap

* Added JSD and rJSD

* Removed JSD and rJSD

* Modified example

* calculateJSD: exprs_values --> abund_values

* all exprs_values --> abund_values

* Revert exprs_values -> abund_values

* Return value distances, renaming

* Added info to man page

* For+for --> apply

* Added tests

* Indentation fix

* version bump, updated news

* Added tests

* Fix

* fix

* Fix

* Added runOverlap

* Version bump

* Renamed the file, resolved man page errors

Co-authored-by: shigdel <86776588+shigdel@users.noreply.github.com>
Co-authored-by: Hervé Pagès <hpages.on.github@gmail.com>
Co-authored-by: Leo Lahti <leo.lahti@iki.fi>
Co-authored-by: Yang Cao <yiluheihei@gmail.com>
Co-authored-by: Sudarshan <sudarshanshetty9@gmail.com>
Co-authored-by: Felix Ernst <felix.gm.ernst@outlook.com>
Co-authored-by: Nitesh Turaga <nitesh.turaga@gmail.com>

FelixErnst commented:
This PR is not aimed at the master branch. Are you aware of that?

TuomasBorman commented:
> This PR is not aimed at the master branch. Are you aware of that?

Yes, that is intentional. This allows, e.g., us to test it before merging into master

TuomasBorman commented:
Maybe best to revert that sync, because this is quite a mess

TuomasBorman committed:
Revert "Issue127 suggestions (#2)" (#3)

This reverts commit d97066c5ff6b440ca2acff3b60ce9d5e5ec764c1.

TuomasBorman committed:
Issue127 new (#4)

* deprecated message

* File name change

TuomasBorman commented on R/calculateUnifrac.R:
```diff
@@ -132,19 +134,38 @@ setMethod("calculateUniFrac",
+        calculateUnifrac(mat, tree = tree, ...)
+    }
+)
+
+#' @rdname calculateUnifrac
+#' @export
+setGeneric("calculateUniFrac", signature = c("x"),
+           function(x, ... )
+             standardGeneric("calculateUniFrac"))
+
+#' @rdname calculateUnifrac
+#' @export
+setMethod("calculateUniFrac",
+          signature = c(x = "ANY"),
+    function(x, ...){
+        .Deprecated( msg = paste0("The name of the function 'calculateUniFrac' is",
+                                  " changed to 'calculateUnifrac'. \nPlease use the new",
+                                  " name instead.\n",
+                                  "See help('Deprecated')") )
+        calculateUnifrac(x, ...)
     }
 )
```
message

TuomasBorman commented:
Branch `issue127_new` includes these changes without conflicts. I'm not sure what would have been the best practice in this case, because now it shows only myself as a contributor

FelixErnst commented:
Can you update from master, to get a current idea of the changes? Thanks

TuomasBorman commented:
How this should be done correctly? 

If I update `shigdel/issue127` from `microbiome/master` over 50 commits appears, because it is not synced. At least last time it showed that all those all 50 commits were new modifications. 

Or should we continue with `issue127_new` branch? This kind of git problems are not too familiar to me

antagomir commented:
If this is just about renaming UniFrac -> Unifrac, perhaps easier to redo all with a clean branch?
